### ISSUE TITLE
Message Input Field Not Cleared When Sending While Japanese IME Candidates Are Displayed

### ISSUE THREAD

**Issue by masarusanjp** [GetStream/stream-chat-swiftui#496]
## What did you do?
I attempted to send a message  while Japanese conversion candidates were still being displayed.

## What did you expect to happen?

I expected the message input field to be cleared after sending the message, even if Japanese conversion candidates were displayed.

## What happened instead?

The message input field was not cleared after sending the message.

## GetStream Environment
**GetStream Chat version:**: 4.56.1
**GetStream Chat frameworks:** StreamChatSwiftUI
**iOS version:** 17.4
**Swift version:** 5.9
**Xcode version:** 15.3
**Device:** iOS Simulator iPhone 15 Pro 

## Additional context

https://github.com/GetStream/stream-chat-swiftui/assets/158416/7c87d219-595b-4f4f-b4f9-1291c008f35e

**Comment by masarusanjp** [comment]
When IME candidates are displayed, markedTextRange becomes non-nil, so it seems that assigning an empty string to InputTextView is not being executed.

https://github.com/GetStream/stream-chat-swiftui/blob/main/Sources/StreamChatSwiftUI/ChatChannel/Composer/ComposerTextInputView.swift#L46 

This issue was resolved with the following fix, but sorry since I have not yet confirmed whether this fix causes any other issues, so I  providing the diff for reference.

```diff
- if uiView.markedTextRange == nil {
+ if uiView.markedTextRange == nil || text.isEmpty {

```

**Comment by martinmitrevski** [comment]
Hi @masarusanjp,

Thanks for reporting, we will have a look. @laevandus an interesting one for you 🇯🇵

**Comment by laevandus** [comment]
Hi @masarusanjp, I checked the proposed fix and we'll include it with the next release. Thank you!

**Comment by masarusanjp** [comment]
Thank you for the quick response and for including the fix in the next release. I'm looking forward to it!

**Comment by laevandus** [comment]
Hi @masarusanjp This issue is fixed in [4.58.0](https://github.com/GetStream/stream-chat-swiftui/releases/tag/4.58.0) Thank you for reporting!
